modules/KoreBuild.Tasks/InstallDotNet.cs

```diff
@@ -0,0 +1,269 @@
+                if (!string.IsNullOrEmpty(request.Channel))
+                {
+                    arguments.Add("-Channel");
+                    arguments.Add(request.Channel);
+                    assetName += $"/{request.Channel}";
+                }
+
+                var expectedPath = request.IsSharedRuntime && !isFloatingVersion
+                    ? Path.Combine(installDir, "shared", "Microsoft.NETCore.App", request.Version, ".version")
+                    : Path.Combine(installDir, "sdk", request.Version, "dotnet.dll");
+
+                if (File.Exists(expectedPath))
+                {
+                    Log.LogMessage(MessageImportance.Normal, $"{assetName} is already installed. Skipping installation.");
+                    continue;
+                }
+
+                Log.LogMessage(MessageImportance.High, $"Installing {assetName}");
+
+                if (isFloatingVersion)
+                {
+                    Log.LogKoreBuildWarning(KoreBuildErrors.DotNetAssetVersionIsFloating, $"The version of {assetName} being installed is a floating version. This may result in irreproducible builds. Consider specifying an exact version number instead.");
+                }
+
+                using (var process = new Process
+                {
+                    StartInfo =
+                    {
+                        FileName = arguments.First(),
+                        Arguments = ArgumentEscaper.EscapeAndConcatenate(arguments.Skip(1)),
+                        RedirectStandardError = true,
+                        RedirectStandardOutput = true,
+                    },
+                    EnableRaisingEvents = true,
+                })
+                {
+                    Log.LogMessage(MessageImportance.Normal, $"Executing {process.StartInfo.FileName} {process.StartInfo.Arguments}");
+
+                    process.OutputDataReceived += LogOutput;
+                    process.ErrorDataReceived += LogError;
+                    var processFinished = new TaskCompletionSource<object>();
+                    process.Exited += (o, e) => processFinished.TrySetResult(null);
+
+                    process.Start();
+                    process.BeginErrorReadLine();
+                    process.BeginOutputReadLine();
+
+                    _cts.Token.Register(() => process.Kill());
+
+                    var timeout = Task.Delay(TimeSpan.FromSeconds(TimeoutSeconds));
+
+                    var finished = await Task.WhenAny(timeout, processFinished.Task);
+
+                    if (ReferenceEquals(finished, timeout))
+                    {
+                        Log.LogError($"dotnet-install of {assetName} timed out after {TimeoutSeconds} seconds");
+                        return false;
+                    }
+
+                    if (process.ExitCode != 0)
+                    {
+                        Log.LogError($"dotnet-install failed on {assetName}. Arguments: {process.StartInfo.FileName} {process.StartInfo.Arguments}");
```

Comment:
There was a bug with powershell where it returned 0 exit code even when script failed. So we have to do `-Command "[System.Threading.Thread]::CurrentThread.CurrentCulture = ''; [System.Threading.Thread]::CurrentThread.CurrentUICulture = '';& '%~dp0build.ps1' %*; exit $LASTEXITCODE"` in `build.cmd`